### PR TITLE
Convert the project to Maven

### PR DESCRIPTION
**Pull request by heruan**
This PR converts the repository structure to a Maven multi-module project. This helps the release process which is currently based on Maven and makes room for additional modules, e.g. the client-side module and a demo project.

The resulting project structure is now:

```
|- observability-kit
\--- observability-kit-agent
```

The `observability-kit-agent` submodule contains the correct Maven dependencies to build and test the project, but to produce the OpenTelemetry agent extension it still contains the Gradle scripts which can be run with the Maven command:

```
mvn -pl :observability-kit-agent package
```

A separate PR will possibly remove the Gradle stuff and use Maven to produce the extended agent.

### CODE REVIEW

**Review by caalador — COMMENTED** [review]
Also if (I would assume it would) the publish goes through maven then it would probably be good to drop `publishing {` block from gradle so it doesn't get published through both by mistake.

File: observability-kit-agent/pom.xml [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.6.2</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>4.7.0</version>
+            <scope>test</scope>
+        </dependency>
```
[the remark this answers is not in the document]
Comment (sissbruecker) [reply observability-kit-agent/pom.xml:68]: Would also have been my question, with the same guess 🙂.

**Comment by sissbruecker** [comment]
If we intend to use the gradle build only for building the agent, but not developing it, we could move / remove all the development related setup:

**Code formatting**

Since the Maven build adds its own formatter plugin, the Spotless code formatter / linter can be removed:
https://github.com/vaadin/observability-kit/blob/75265a0f44772f88a6b588491351ae96812090f0/observability-kit-agent/build.gradle#L15
https://github.com/vaadin/observability-kit/blob/75265a0f44772f88a6b588491351ae96812090f0/observability-kit-agent/build.gradle#L200-L206

**Unit tests**

Tests should probably be run through Maven as well? Currently running `mvn test` from the agent module does not seem to pick up any tests. If running tests has been added to the Maven config, then the test related setup can be removed from Gradle:

https://github.com/vaadin/observability-kit/blob/75265a0f44772f88a6b588491351ae96812090f0/observability-kit-agent/build.gradle#L27
https://github.com/vaadin/observability-kit/blob/75265a0f44772f88a6b588491351ae96812090f0/observability-kit-agent/build.gradle#L33
https://github.com/vaadin/observability-kit/blob/75265a0f44772f88a6b588491351ae96812090f0/observability-kit-agent/build.gradle#L103-L115

File: observability-kit-agent/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (heruan) [reply observability-kit-agent/pom.xml:68]: Yes, exactly! I'm still trying to remove Gradle completely before merging this 😅

**Pull request merged**